### ISSUE TITLE
axis with int64 is not supported by tflite for strided_slice op

### ISSUE THREAD

**Issue by snsun** [tensorflow/tensorflow#44407]
Hi,
Recentlly, I converted my pytorch model to pb using onnx-tf. I can convert successfully. But during inference, I found that in my pb model, the axis of strided_slice is int64. How can I convert int64 to int32 axis?

Here is the error:
> ERROR: tensorflow/lite/kernels/strided_slice.cc:155 op_context.begin->type != kTfLiteInt32 (4 != 2)
> ERROR: Node number 22 (STRIDED_SLICE) failed to prepare.

https://github.com/playbar/tfcmake/blob/e5c8319827fd06146f1edcc1557f0189e50da89d/tensorflow/contrib/lite/kernels/strided_slice.cc#L157

I think this because the axis in slice op is int64 and tflite does not support int64.  Is there a method to solve this problem?

 - Python version: 3.6
 - ONNX version:   1.7
 - ONNX-TF version: 1.6
 - Tensorflow version: 1.15

**Comment by ravikyram** [comment]
@snsun 

Can you please refer #11318, [SO link ](https://stackoverflow.com/questions/47215739/operation-has-type-int32-that-does-not-match-type-int64) and see if it helps you.Thanks!

**Comment by ravikyram** [comment]
@snsun 
Please, refer this #11318, [SO link ](https://stackoverflow.com/questions/47215739/operation-has-type-int32-that-does-not-match-type-int64)and see if it helps you.

If the problem still persists please share the exact sequence of commands / steps that you executed before running into the problem. So we will try to reproduce the issue from our end to localize the issue faster. Thanks!

**Comment by snsun** [comment]
Thanks. I solved this problem by modifying the onnx-tf code. When onnx-tf is used to convert onnx model to pb model, the begin and end indices are int64. I cast them to int32 and the problem is solved. 
By the way, I noticed a TODO in https://github.com/playbar/tfcmake/blob/e5c8319827fd06146f1edcc1557f0189e50da89d/tensorflow/contrib/lite/kernels/strided_slice.cc#L156
I am curious why int64 is not supported now for strided_slice op.

**Comment by jvishnuvardhan** [comment]
@snsun Casting is right approach here. 

Currently int64 is not supported. Generally most of the use-cases are implemented on device and want to shrink the model size. For those use-cases, int64 is an overhead in terms of size/memory and speed. 

If more users are looking for int64 support, then our team will prioritize it and will implement it. Thanks!

I am closing this issue as this was resolved. Please feel free to reopen if I am mistaken. Thanks!

**Comment by junhyk** [comment]
@jvishnuvardhan 
Hi, 
i'm using RaggedTensor for my model.
When I converted the model into tflite, I found there is whole lot of FlexStridedSlice in the computation graph like below
![image](https://user-images.githubusercontent.com/26551254/98449118-528fa280-2174-11eb-850c-fa838d025261.png)

Is this because tflite's stridedslice doesn't support int64?
If true, can I cast ragged tensor's row_split into int32?

Thanks

**Comment by hubbup** [comment]
> Thanks. I solved this problem by modifying the onnx-tf code. When onnx-tf is used to convert onnx model to pb model, the begin and end indices are int64. I cast them to int32 and the problem is solved.
> By the way, I noticed a TODO in https://github.com/playbar/tfcmake/blob/e5c8319827fd06146f1edcc1557f0189e50da89d/tensorflow/contrib/lite/kernels/strided_slice.cc#L156
> I am curious why int64 is not supported now for strided_slice op.

I have the same issue. Can I ask you how you modified your onnx-tf code to cast?

**Comment by hubbup** [comment]
@snsun 
> Thanks. I solved this problem by modifying the onnx-tf code. When onnx-tf is used to convert onnx model to pb model, the begin and end indices are int64. I cast them to int32 and the problem is solved.
> By the way, I noticed a TODO in https://github.com/playbar/tfcmake/blob/e5c8319827fd06146f1edcc1557f0189e50da89d/tensorflow/contrib/lite/kernels/strided_slice.cc#L156
> I am curious why int64 is not supported now for strided_slice op.

I have the same problem. Can you let me know how I have to modify the onnx-tf code to cast?

**Comment by zhihongp** [comment]
> @snsun
> 
> > Thanks. I solved this problem by modifying the onnx-tf code. When onnx-tf is used to convert onnx model to pb model, the begin and end indices are int64. I cast them to int32 and the problem is solved.
> > By the way, I noticed a TODO in https://github.com/playbar/tfcmake/blob/e5c8319827fd06146f1edcc1557f0189e50da89d/tensorflow/contrib/lite/kernels/strided_slice.cc#L156
> > I am curious why int64 is not supported now for strided_slice op.
> 
> I have the same problem. Can you let me know how I have to modify the onnx-tf code to cast?

Same problem here and wonder how this is done

**Comment by LYXwizard** [comment]
> @snsun
> 
> > Thanks. I solved this problem by modifying the onnx-tf code. When onnx-tf is used to convert onnx model to pb model, the begin and end indices are int64. I cast them to int32 and the problem is solved.
> > By the way, I noticed a TODO in https://github.com/playbar/tfcmake/blob/e5c8319827fd06146f1edcc1557f0189e50da89d/tensorflow/contrib/lite/kernels/strided_slice.cc#L156
> > I am curious why int64 is not supported now for strided_slice op.
> 
> I have the same problem. Can you let me know how I have to modify the onnx-tf code to cast?

Same problem here and wonder how this is done
